### ISSUE TITLE
EOFError: encountered eof while reading data (for cyphers of more then 255 symbols)

### ISSUE THREAD

**Issue by sergeonito** [neo4jrb/neo4j-ruby-driver#168]
Hi,
﻿﻿I am getting an **EOFError: encountered eof while reading data error**, when run cypher query with more than 255 symbols
﻿﻿
Some of backtrace is below:
```
               |   → ./async-io-1.33.0/lib/async/io/stream.rb:114 in `read_exactly'
               |     ./neo4j-ruby-driver-4.4.0.alpha.2/ruby/neo4j/driver/internal/async/inbound/chunk_decoder.rb:35 in `read_length_field'
               |     ./neo4j-ruby-driver-4.4.0.alpha.2/ruby/neo4j/driver/internal/async/inbound/chunk_decoder.rb:13 in `read_exactly'`
               |     ./neo4j-ruby-driver-4.4.0.alpha.2/ruby/neo4j/driver/internal/packstream/pack_input.rb:10 in `read_byte'
               |     ./neo4j-ruby-driver-4.4.0.alpha.2/ruby/neo4j/driver/internal/packstream/pack_stream.rb:221 in unpack_struct_header'
```
               
For queries less then 255 symbols all works fine

**ENV:**
  ruby-3.1.2
  rails 6.1.2
  neo4j-ruby-driver 4.4.0.alpha.2
  neo4j server 4.4.8 enterprise

**Example:**
```
query = <<~QUERY
  MATCH (node1:Node1)
  MATCH (node2:Node2)
  MATCH (node3:Node3)
  MATCH (node4:Node4)
  MATCH (node5:Node5)
  MATCH (node6:Node6)
  MATCH (node7:Node7)
  MATCH (node8:Node8)
  MATCH (node9:Node9)
  MATCH (node10:Node10)
  MATCH (node11:Node11)
  MATCH (node12:Node12)
  RETURN node1, node2, node3
QUERY
  
require 'neo4j_ruby_driver'

$neo4j = Neo4j::Driver::GraphDatabase.driver(
  ENV['NEO4J_URL'],
  Neo4j::Driver::AuthTokens.basic(ENV['NEO4J_USERNAME'], ENV['NEO4J_PASSWORD'])
)

$neo4j.session.run(query)
```

**Comment by klobuczek** [comment]
@sergeonito thanks! Reproduced. Interesting that queries of size 2**16 work again. :)

**Comment by trialize** [comment]
Looks like a magic. Any ideas how to address or at least where to dig into?

**Comment by klobuczek** [comment]
Released neo4j-ruby-driver (4.4.0.alpha.3)
